src/js/CheckboxTree.js

```diff
@@ -212,64 +235,152 @@ class CheckboxTree extends React.Component {
             showNodeTitle,
             showNodeIcon,
         } = this.props;
-        const { id, model } = this.state;
-        const { icons: defaultIcons } = CheckboxTree.defaultProps;
 
+        const { icons, id } = this.state;
+
+        let state1counter = 0;
+        let state2counter = 0;
         const treeNodes = nodes.map((node) => {
+            this.parents[node.value] = parent;
+
             const key = node.value;
-            const flatNode = model.getNode(node.value);
-            const children = flatNode.isParent ? this.renderTreeNodes(node.children, node) : null;
+            const isParent = this.isParent(node);
+            const isRadioGroup = !!node.radioGroup;
+            const isRadioNode = !!parent.radioGroup;
+
+            /*
+            //---------------------------------------------------------------
+            // this checks for multiple checked === true nodes in a RadioGroup
+            // This fixes the problem by mutating the prop!
+            if (isRadioGroup) {
+                const numChecked = node.children.filter(child => child.checked).length;
+                if (numChecked !== 1) {
+                    // set checked = true for first child as default
+                    const defaultChecked = 0;
+                    for (let i = 0, ii = node.children.length; i < ii; i += 1) {
+                        // esLint-disable-next-line no-param-reassign
+                        node.children[i].checked = (i === defaultChecked);
+                    }
+                }
+            }
+            //---------------------------------------------------------------
+            */
+
+            // determine if node needs to be disabled
+            let nodeDisabled = disabled || node.disabled || forceDisabled;
+            if ((!noCascade && parent.disabled) ||
+                (isRadioNode && !parent.checked)
+            ) {
+                nodeDisabled = true;
+            }
 
-            // Determine the check state after all children check states have been determined
-            // This is done during rendering as to avoid an additional loop during the
-            // deserialization of the `checked` property
-            flatNode.checkState = this.determineShallowCheckState(node, noCascade);
+            // determine if node children need to be disabled
+            let disableChildren = false;
+            if ((!noCascade && nodeDisabled) || (isRadioNode && !node.checked)) {
```

Comment:
This check for a parent node being checked before radio options may be selected could present an issue if the parent has `showCheckbox: false`, as the parent could never be selected. Otherwise I think it makes sense.

Reply:
Thanks for pointing this out. You have caught a bug.

My thinking on this is that a `radioGroup` node, the parent of the 'radio button' nodes, should always have a checkbox. It works as a leaf in the tree to that point. Each radio button below it works as the start of another tree. 

We probably need to work through some further discussion on this as from some of your comments I see you have a different use case using the `checked` array.  I will start that discussion in a new comment at the bottom.